lib/schema.js

```diff
@@ -13,119 +19,80 @@ const SCOPE = ['application', 'global', 'local'];
 const STORE = ['persistent', 'memory'];
 const ALLOW = ['write', 'append', 'read'];
 
-const ES_TYPES = ['number', 'string', 'boolean'];
+const ES_TYPES = ['number', 'string', 'boolean', 'bigint'];
 
-const checks = {
-  array: (arr, type) => {
-    if (!Array.isArray(arr)) return false;
-    for (const el of arr) {
-      if (typeof el !== type) return false;
-    }
-    return true;
-  },
-
-  set: (set, type) => {
-    if (Reflect.getPrototypeOf(set) !== Set.prototype) return false;
-    for (const el of set.values()) {
-      if (typeof el !== type) return false;
-    }
-    return true;
-  },
-
-  object: (obj, key, val) => {
-    if (typeof obj !== 'object') return false;
-    for (const [name, value] of Object.entries(obj)) {
-      if (typeof name !== key) return false;
-      if (typeof value !== val) return false;
-    }
-    return true;
-  },
-
-  map: (map, key, val) => {
-    if (Reflect.getPrototypeOf(map) !== Map.prototype) return false;
-    for (const [name, value] of map.entries()) {
-      if (typeof name !== key) return false;
-      if (typeof value !== val) return false;
-    }
-    return true;
-  },
-};
+const RESERVED_FOR_SCHEMA = ['validate', 'parse', 'serialize', 'format'];
+const RESERVED_FOR_DEF = ['required', 'unique', 'default', 'note', 'index'];
 
-const getShorthand = (def) => {
-  const keys = Object.keys(def);
-  for (const key of keys) {
-    if (def[key].shorthand) return key;
+const setValidate = (validate) => {
+  if (!validate) return () => ({ valid: true, errors: [] });
+  if (typeof validate !== 'function') {
+    throw new Error('Validate must be a function');
   }
-  return '';
+  return (value, path) => {
```

Comment:
This closure is duplicated between Schemas (and therefore uses more memory), wdyt of having a separate function and call .bind instead?